src/main/java/org/springframework/integration/kafka/listener/KafkaTopicOffsetManager.java

```diff
@@ -0,0 +1,505 @@
+	 */
+	public void setMaxQueueBufferingTime(int maxQueueBufferingTime) {
+		this.maxQueueBufferingTime = maxQueueBufferingTime;
+	}
+
+	/**
+	 * The size of a segment in the offset topic
+	 *
+	 * @param segmentSize the segment size of an offset topic
+	 */
+	public void setSegmentSize(int segmentSize) {
+		this.segmentSize = segmentSize;
+	}
+
+	/**
+	 * How long are dead records retained in the offset topic
+	 *
+	 * @param retentionTime the retention time for dead records (in seconds)
+	 */
+	public void setRetentionTime(int retentionTime) {
+		this.retentionTime = retentionTime;
+	}
+
+	/**
+	 * The replication factor of the offset topic
+	 *
+	 * @param replicationFactor the replication factor
+	 */
+	public void setReplicationFactor(int replicationFactor) {
+		this.replicationFactor = replicationFactor;
+	}
+
+	/**
+	 * The maximum batch size for offset writes
+	 *
+	 * @param maxBatchSize maximum batching window
+	 */
+	public void setMaxBatchSize(int maxBatchSize) {
+		this.maxBatchSize = maxBatchSize;
+	}
+
+	/**
+	 * Whether offset writes should be batched or not
+	 *
+	 * @param batchWrites true if writes are batched
+	 */
+	public void setBatchWrites(boolean batchWrites) {
+		this.batchWrites = batchWrites;
+	}
+
+	/**
+	 * The number of required acks on write operations
+	 *
+	 * @param requiredAcks the number of required acks
+	 */
+	public void setRequiredAcks(int requiredAcks) {
+		this.requiredAcks = requiredAcks;
+	}
+
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		ZkClient zkClient = new ZkClient(zookeeperConnect.getZkConnect(),
+				Integer.parseInt(zookeeperConnect.getZkSessionTimeout()),
+				Integer.parseInt(zookeeperConnect.getZkConnectionTimeout()),
+				ZKStringSerializer$.MODULE$);
+
+		try {
+			createCompactedTopicIfNotFound(zkClient);
+			validateOffsetTopic(zkClient);
+			Partition offsetPartition = new Partition(topic, 0);
+			BrokerAddress offsetPartitionLeader = connectionFactory.getLeader(offsetPartition);
+			readOffsetData(offsetPartition, offsetPartitionLeader);
+			initializeProducer(offsetPartitionLeader);
+		}
+		finally {
+			try {
+				zkClient.close();
+			}
+			catch (ZkInterruptedException e) {
+				log.error("Error while closing Zookeeper client", e);
+			}
+		}
+	}
+
+	@Override
+	protected void doUpdateOffset(Partition partition, long offset) {
+		data.put(partition, offset);
+		producer.send(new KeyedMessage<Key, Long>(topic, new Key(consumerId, partition), offset));
+	}
+
+	@Override
+	protected void doRemoveOffset(Partition partition) {
+		data.remove(partition);
+		producer.send(new KeyedMessage<Key, Long>(topic, new Key(consumerId, partition), null));
+	}
+
+	@Override
+	protected Long doGetOffset(Partition partition) {
+		return data.get(partition);
+	}
+
+	@Override
+	public void flush() throws IOException {
+		// not supported
+	}
+
+	@Override
+	public void close() throws IOException {
+		producer.close();
```

Comment:
Shouldn't this class be `DisposableBean` or Spring takes care about `Closeable` well ?